core/src/main/java/io/undertow/server/handlers/accesslog/DefaultAccessLogReceiver.java

```diff
@@ -243,43 +235,73 @@ void awaitWrittenForTest() throws InterruptedException {
         }
     }
 
-    private void writeMessage(final List<String> messages) {
-        if (System.currentTimeMillis() > changeOverPoint) {
-            doRotate();
+    private void writeMessage(final String message) {
+        //NOTE: is there a need to rotate on write?
+        //if (System.currentTimeMillis() > changeOverPoint) {
```

Comment:
Yes, but only once per run. Now that this method is called per message it is no longer the adequate place for that.

Reply:
Probably we should add the limit to 1000 here? I think it would perform a bit better.

Reply:
I still stand my ground, I think the granularity of checking 1000 here performs better.